server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt

```diff
@@ -109,133 +117,170 @@ object Chapter {
+                            .toList()
+                    }
 
-        val chaptersToInsert = mutableListOf<ChapterDataClass>()
-        val chaptersToUpdate = mutableListOf<ChapterDataClass>()
-
-        chapterList.reversed().forEachIndexed { index, fetchedChapter ->
-            val chapterEntry = chaptersInDb.find { it.url == fetchedChapter.url }
-
-            val chapterData =
-                ChapterDataClass.fromSChapter(
-                    fetchedChapter,
-                    chapterEntry?.id ?: 0,
-                    index + 1,
-                    now,
-                    mangaId,
-                    runCatching {
-                        (source as? HttpSource)?.getChapterUrl(fetchedChapter)
-                    }.getOrNull(),
-                )
-
-            if (chapterEntry == null) {
-                chaptersToInsert.add(chapterData)
-            } else {
-                chaptersToUpdate.add(chapterData)
-            }
-        }
+                val chaptersToInsert = mutableListOf<ChapterDataClass>()
+                val chaptersToUpdate = mutableListOf<ChapterDataClass>()
+
+                chapterList.reversed().forEachIndexed { index, fetchedChapter ->
+                    val chapterEntry = chaptersInDb.find { it.url == fetchedChapter.url }
+
+                    val chapterData =
+                        ChapterDataClass.fromSChapter(
+                            fetchedChapter,
+                            chapterEntry?.id ?: 0,
+                            index + 1,
+                            now,
+                            mangaId,
+                            runCatching {
+                                (source as? HttpSource)?.getChapterUrl(fetchedChapter)
+                            }.getOrNull(),
+                        )
 
-        transaction {
-            if (chaptersToInsert.isNotEmpty()) {
-                ChapterTable.batchInsert(chaptersToInsert) {
-                    this[ChapterTable.url] = it.url
-                    this[ChapterTable.name] = it.name
-                    this[ChapterTable.date_upload] = it.uploadDate
-                    this[ChapterTable.chapter_number] = it.chapterNumber
-                    this[ChapterTable.scanlator] = it.scanlator
-                    this[ChapterTable.sourceOrder] = it.index
-                    this[ChapterTable.fetchedAt] = it.fetchedAt
-                    this[ChapterTable.manga] = it.mangaId
-                    this[ChapterTable.realUrl] = it.realUrl
+                    if (chapterEntry == null) {
+                        val newChapterData =
+                            if (chapterData.uploadDate == 0L) {
+                                val altDateUpload = if (maxSeenUploadDate == 0L) now else maxSeenUploadDate
+                                chapterData.copy(uploadDate = altDateUpload)
+                            } else {
+                                maxSeenUploadDate = max(maxSeenUploadDate, chapterData.uploadDate)
```

Comment:
not needed due to line 189

Reply:
Thanks, the one on 189 isn't supposed to be there.